### DISCUSSION TITLE
Lernorte: Konzeptuelle Überarbeitung

### DISCUSSION THREAD

**Discussion by mstuder** [fluxapps/ILIAS-Pegasus#31] [Feature Requests]
Aufgrund des Fehlers - https://github.com/fluxapps/ILIAS-Pegasus/issues/28 - habe ich mich intensiv mit den Lernorten auseinandergesetzt und möchte ein paar konzeptuelle Änderungen mit euch diskutieren. Aus meiner Sicht müssen wir derzeitige Komplexität bei der Umsetzung reduzieren, damit der Fehler behoben werden kann und künftig Fehler besser nachvollzogen werden können.

**1. Lernorte als besucht markieren**
Ich schlage vor, dass ein Lernort explizit als "besucht" markiert werden soll. Es soll somit kein Hintergrundsprozess laufen, welcher prüft, ob man beim Lernort ist. Sondern der User soll diesen per Button als besucht markieren.

Hierfür wird eine neue Schaltfläche eingeführt:
"Try to mark as visited"
![image](https://user-images.githubusercontent.com/472326/194777015-de31e30e-6913-4f35-89e5-b747ae9de5c4.png)

Ist der Benutzer zu weit vom Lernort weg, wird ihm dies als Rückmeldung angzeigt.

Ist der Benutzer direkt beim Lernort
... so werden die entsprechenden Inhalte "Visible after visit",  "Visible at place" freigeschalten
... zudem werden werden zwei Status-Icons (nicht in obiger Abbildung ersichtlich)  "Visible after visit", "Visible at place - 30 min"  auf OK geschalten. Der Status _Visible at place_ wird bei jedem neuen Lernortwechsel neu geprüft, sind 30 Minuten um, müsste der User wiederum die "Try to mark as visited" Schaltfläche klicken.

Bin froh, über Vorschläge eines besseren Labels für die Schaltfläche.

**2. Lernorte mittels QR Code als besucht markieren**
Künftig möchten wir ermöglichen, dass Lernorte mittels QR-Code als besucht markiert werden können. In diesem Fall würde beim Klick obiger Schaltfläche die Kamera mittels QR-Code-Reader gestartet.

**3. Karte / markierter Punkt**
Auf der Karte wird ausschliesslich der aktuell gewählte Lernort angezeigt. Möchte man einen anderen Punkt auf der Karte anzeigen lassen, wechselt man mittels Navigation den markierten Punkte auf der Karte.
![image](https://user-images.githubusercontent.com/472326/194778023-e14a00b3-7d60-477f-a3ba-235f33bfe9a4.png)

**4. Lernorte Plugin - verzichten auf globale Einstelllung**
Ich verstehe nicht, weshalb wir hier eine zur Location zusätzliche Karte haben:
![image](https://user-images.githubusercontent.com/472326/194777689-c9ca0be7-97ae-4930-b34a-6982c52a04a6.png)

Zudem frage ich mich, ob wir auf die globale Einstellung, ob der Lernort sichtbar ist oder nicht gänzlich verzichten können:
![image](https://user-images.githubusercontent.com/472326/194777732-b13973d3-7c25-462c-87fd-e68acbb4735d.png)

Dies weil ja nun die einzelnen Inhalte als sichtbar / unsichtbar markiert werden.

Wie ist eure Meinung dazu?

**Comment by mglaubitz** [comment]
Hallo Martin,
vielen Dank für die Einarbeitung und die Vorschläge. Ich habe zwei Anmerkungen dazu:

ad 2. Lernorte mittels QR Code als besucht markieren:
ich glaube,dass es hierfür einen extra Button braucht, da die App ja nicht weiß, dass ich nun hier gerade per QR Code einchecken will . Ich glaube ein Button mit einer QR-Glyphe neben dem Button ("Try mark as visited" / oder "Try to check-in") zum Einchekcen würde sich sehr gut eignen. 

ad 3. Karte / markierter Punkt:
wegen mir kann auf der Karte, die man sieht, wenn man auf einen Lernort klcik auch nur genau der Lernort angezeigt werden. Was wir braucehn ist aber eine Karte, auf der man alle Lernorte im aktuellen Container sieht. Die aktuelle Umsetzung war da nur ein Kompromiss. Wegen mir kann die "Karten-Glyphe", die man derzeit nur innerhalb von Lernorten sieht, auch immer dann angeziegt werden, wenn im aktuellen Container mind. ein Lernort liegt - praktisch dann neben dem Sternchen für die Favoriten

**Comment by rfcmaXi** [comment]
Hallo Martin,
vielen dank für die Mühe und Vorschläge!

**3. Karte / markierter Punkt**
Ich schließe mich hier an. Wenn es darum geht, die Last zu verringern: Bitte nur den aktuellen Lernort in der Karte anzeigen! Sehe es aber hinsichtlich der Gesamtkarte genau wie Marco. Im Container-Objekt brauchen wir diese zwingend. 

- Container -> Gesamtkarte mit allen Lernorte
- am Lernort selbst -> Karte nur mit Lernort selbst

**2. Lernorte mittels QR Code als besucht markieren**
Schließe mich Marcos Ansatz mit dem Button + QR-Code-Symbol, welches die Kamera startet, an.

**1. Lernorte als besucht markieren**
Finde den Ansatz auch okay. Die Last würde sich verringern und der Studi müsste eine bewusste Entscheidung treffen. Marco hat bei QR-Codes vom einchecken gesprochen. Vielleicht ist dies auch ein klareres Wort als "besucht"? Man checkt man Lernort ein - egal ob per GPS oder per QR-Code. 

Grüße!

**Comment by mstuder** [comment]
Besprechung Marko / Martin

**Lernorte als besucht markieren**
- check-in (einchecken)
-- eingescheckt
-- nicht eingecheckt, da zu weit weg
- QR-Code 

ein check-in ist 30 min gültig für Inhalte, welche nur am Lernort sichtbar sind. Contents, welche bereits geöffnet sind, sind davon nicht betroffen.

checkin-button wird grün, sobald betätigt und bleibt 30 minuten grün. Nach den 30 Minuten wird der Button zu "checkin again"

**Karte**
Funktion zur Sichtbarkeit der Karte wird entfernt. Karte wird im Subtab dargestellt aufgrund der Location-Einstellung.

**App**
Ein Lernplace in der App zeigt nur noch den expliziten Learnplace an. Zusätzlich wird eine Übersichtskarte dargestellt mit allen Karten.
-> In der Liste _Show map with all Learnplaces_

**Comment by mhilbert** [comment]
**zu 1. Lernorte als besucht markieren**

Wenn es die Performance erhöht, bin ich dafür einen "Check-In"-Button einzuführen. Ggf. wäre alternativ die Frage, ob es nicht auch ausreichen würde, das Abfrageintervall zu vergrößern?

Nach 30 Minuten sollte allerdings nur dann der Status zurückgesetzt werden, wenn sich der Nutzer tatsächlich vom Lernort entfernt hat. D.h. die App sollte nach Ablauf der 30 Minuten eine GPS-Abfrage machen, ob der Nutzer sich noch innerhalb des definierten Radius zum Lernort befindet. Wenn ja, sollte die Zeit um weitere 30 Minuten verlängert werden. Wenn nein, sollte der Status zurückgesetzt werden, so dass der Nutzer beim erneuten Besuch erneut "Einchecken" muss.

Eine Ausnahme wären Indoor-Lernorte, zu denen möglicherweise gar keine GPS-Koordinaten hinterlegt sind, und eine GPS-Abfrage somit gar nicht möglich ist (siehe 2.). Eine Zeitverlängerung wäre in diesem Ausnahmefall nicht möglich.

Die Zeit (Default: 30 Minuten) sollte in der Plugin-Konfiguration des Lernplaces-Plugin konfigurierbar sein. Besser noch in den Einstellungen des Lernortes, um unterschiedliche didaktische Szenarien zu unterstützen.

Den Begriff "Check-In" empfinde ich als unpassend. Hier ein paar Vorschläge für die Beschriftung des Check-In-Buttons:
- "Bin da" | "Bin wieder da" oder "Da bin ich wieder"
- "Bin hier" | "Bin wieder hier"
- "Bin vor Ort" | "Bin wieder vor Ort"
- "Bin am Lernort" | "Bin wieder am Lernort"
- "Bin angekommen" | "Bin wieder angekommen"



**zu 2. Lernorte mittels QR Code als besucht markieren**

Ein "Check-In" per QR-Code-Scan, wie hier vorgeschlagen, ist umständlich (siehe Beispiel: Variante 3). Der QR-Code-Scan sollte direkt über die Hauptnavigation der Pegasus-App aufrufbar sein, nicht über den "Check-In"-Button des Lernorts.

_Beispiel:_ ein Gemälde von Van Gogh in der Nationalgalerie in London als Indoor-Lernort (versehen mit einem QR-Code, da GPS-Navigation im Gebäude nicht möglich)
- _Variante 1 (komfortabel):_
 Der Lerner
 -- öffnet seine Foto-App auf dem Smartphone
 -- und scannt den QR-Code.
 => Es öffnet sich die Pegasus-App, der ausgewählte Lernort wird angezeigt, ein Check-In ist automatisch erfolgt.
- _Variante 2 (auch gut):_
 Der Lerner
 -- öffnet die Pegasus-App auf dem Smartphone,
 -- tippt auf die Schaltfläche "QR-Code" in der Hauptnavigation
 -- und scannt den QR-Code.
 => Der ausgewählte Lernort wird angezeigt, ein Check-In ist automatisch erfolgt.
- _Variante 3 (umständlich):_
 Der Lerner
 -- öffnet die Pegasus-App auf dem Smartphone,
 -- sucht in seinen Kursen/Gruppen/Ordnern nach dem Lernort zu dem Bild, vor dem er gerade steht,
 -- öffnet den Lernort,
 -- tippt auf den "Check-In"-Button
 -- und scannt den QR-Code.
 => Das Check-In wird durchgeführt und ggf. versteckte Inhalte erscheinen.

Siehe Feature-Eintrag "QR Code Scan in der App für alle Objekte und speziell Lernorte"
https://docu.ilias.de/goto_docu_wiki_wpage_7581_6052.html



**zu 3. Karte / markierter Punkt**

In der Kartenansicht zu einem Lernort reicht es aus, den Marker zum ausgewählten Lernort anzuzeigen. Ein Wechseln der Anzeige ist aus meiner Sicht nicht nötig bzw. verwirrend, wenn in der Karte der Marker zu Lernort Y angezeigt wird, obwohl man gerade die Inhalte zu Lernort X angezeigt bekommt.

Eine Ansicht aller Lernorte im aktuellen Container sollte unabhängig davon möglich sein, siehe Feature-Eintrag "Kartenansicht aller Lernorte auf Kursebene"
https://docu.ilias.de/goto_docu_wiki_wpage_6110_6052.html



**zu 4. Lernorte Plugin - verzichten auf globale Einstellung**

Die Darstellung der Karte im Unterreiter "Karte" unterhalb des Reiters "Inhalt" ist eigentlich überflüssig, da die Karte mit dem Marker bereits im Reiter "Einstellungen" unter dem Punkt "Position" dargestellt wird. Der Unterreiter dient allerdings der Einstellungsmöglichkeit der ortsabhängigen Sichtbarkeit der Karte.

Solange aber die ortsabhängige Sichtbarkeit der Position des Lernorts nicht einstellbar ist, sollte die Einstellung der ortsabhängigen Sichtbarkeit der Karte weiter möglich sein, um das Szenario von "Hidden Places" weiter bedienen zu können:
Analog zum Geocaching oder zu Outdoor-Escape-Games muss ein Lernort anhand von bestimmten Hinweisen erst gefunden werden. D.h. die Position des Lernorts sollte bis zum Erreichen des Lernorts unbekannt sein. Erst wenn man am Lernort angekommen ist (und sich erfolgreich einchecken konnte), sollte die Karte (bzw. neu: der Positionsmarker in der Karte) angezeigt werden.

Siehe Feature-Eintrag "Hinzufügen einer Karte bzw. Einstellung der Sichtbarkeit", Vorschlag 2:
https://docu.ilias.de/goto_docu_wiki_wpage_6049_6052.html

Hier ein erweiterter Feature-Eintrag zum Szenario "Hidden Places", der nicht nur die Position des Lernorts im Dunklen lässt, sondern auch den Titel, indem das komplette Lernort-Objekt solange unsichtbar bleibt, bis die angegebene Position erreicht wurde.
Dieses Szenario ist aber mit der Notwendigkeit eines Check-Ins nicht mehr abbildbar.
https://docu.ilias.de/goto_docu_wiki_wpage_6111_6052.html

**Comment by rfcmaXi** [comment]
Hallo Mirco,

wir haben folgendes heute beim Meeting besprochen:

zu 1. Lernorte als besucht markieren:
Wir wollen Komplexität verlieren. Daher haben wir uns entschieden, keinen automatischen Check nach 30 Minuten zu machen. Wir können den Ansatz aber total verstehen.
Hinsichtlich der Begrifflichkeit sollten wir noch mal zusammen diskutieren. Alle deine Ideen sind gut.

zu 2. Lernorte mittels QR Code als besucht markieren
Da hast du eine harte Diskussion ausgelöst... Martin hat dazu ein Flussdiagramm erstellt, ich denke, er postet es hier rein.

zu 3. Karte / markierter Punkt
Wird so gemacht!

zu 4. Lernorte Plugin - verzichten auf globale Einstellung
Wir wollen Komplexität verlieren - again. D.h. wir nehmen die Karte aus den Optionen raus, sind uns aber bewusst, dass wir damit ggf. auch versteckte Lernorte verlieren und in einem späteren Aufschlag erneuet, bei Bedarf, erweitern würden.

Grüße und Danke für das Feedback!!
